File: factory/Exemplo1/TesteApp.java
public class TesteApp {
    public static void main(String[] args) {
        AnimalFactory fabCaes = new AnimalCachorroImpl();
		AnimalFactory fabGatos = new AnimalGatoImpl();

        String nome = "James";
		Animal dog = fabCaes.getAnimal(nome);
        dog.falar();
        
        nome = "Marvin";
        Animal cat = fabGatos.getAnimal(nome);
        cat.falar();
    }
}

File: factory/Exemplo1/TesteApp.ts
class TesteApp {
    public static main(): void {
        const fabCaes: AnimalFactory = new AnimalCachorroImpl();
        const fabGatos: AnimalFactory = new AnimalGatoImpl();

        let nome = "James";
        const dog: Animal = fabCaes.getAnimal(nome);
        dog.falar();

        nome = "Marvin";
        const cat: Animal = fabGatos.getAnimal(nome);
        cat.falar();
    }
}

// Executar o método main
TesteApp.main();
